src/commands/test.js

```diff
@@ -90,3 +141,109 @@ function getPlatform(logs) {
 
   throw new Error(`Unsupported platform: ${type} ${arch} ${majorVersion}`)
 }
+
+function runDocker(coverage_opt, datasource, version_opt, latest_version, print) {
+  // Remove binary-install-raw binaries, because docker has permission issues
+  // when building the docker images
+  fs.rmSync("node_modules/binary-install-raw/bin", { force: true, recursive: true });
+
+  let dir = 'tests/.docker';
+
+  if (!fs.existsSync(dir)){
+    fs.mkdirSync(dir, { recursive: true });
+  }
+
+  try {
+    fs.writeFileSync(`${dir}/Dockerfile`, dockerfile(version_opt, latest_version))
+    print.info('Successfully generated Dockerfile.');
+  } catch (error) {
+    print.info('A problem occurred while generating the Dockerfile. Please attend to the errors below:');
+    print.info(chalk.red(error));
+    return
+  }
+
+  // Run a command to check if matchstick image already exists
+  exec('docker images -q matchstick', (error, stdout, stderr) => {
+    // Getting the current working folder that will be passed to the
+    // `docker run` command to be bind mounted.
+    let current_folder = process.cwd();
+    let test_args = '';
+
+    if(datasource) {
+      test_args = test_args + datasource
+    }
+
+    if(coverage_opt) {
+      test_args = test_args + ' ' + '-c'
+    }
+
+    let docker_run_opts = ['run', '-it', '--rm', '--mount', `type=bind,source=${current_folder},target=/matchstick`];
+
+    if(test_args !== '') {
+      docker_run_opts.push('-e')
+      docker_run_opts.push(`ARGS=${test_args.trim()}`);
+    }
+
+    docker_run_opts.push('matchstick')
+
+    // If a matchstick image does not exists, the command returns an empty string,
+    // else it'll return the image ID. Skip `docker build` if an image already exists
+    // If `-v/--version` is specified, delete current image(if any) and rebuild.
+    // Use spawn() and {stdio: 'inherit'} so we can see the logs in real time.
+    if(stdout === '' || version_opt) {
+      if (stdout !== '' && version_opt) {
+        exec('docker image rm matchstick', (error, stdout, stderr) => {
+          print.info(chalk.bold(`Removing matchstick image\n${stdout}`));
+        });
+      }
+      // Build a docker image. If the process has executed successfully
+      // run a container from that image.
+      spawn(
+        'docker',
+        ['build', '-f', 'tests/.docker/Dockerfile', '-t', 'matchstick', '.'],
+        { stdio: 'inherit' }
+      ).on('close', code => {
+        if (code === 0) {
+           spawn('docker', docker_run_opts, { stdio: 'inherit' });
+        }
+      })
+    } else {
+      // Run the container from the existing matchstick docker image
+      spawn('docker', docker_run_opts, { stdio: 'inherit' });
+    }
+  })
+}
+
+// TODO: Move these in separate file (in a function maybe)
```

Comment:
This is a nice TODO, I don't see as a requirement, but you can definitely do it 😊 

We do it for our `docker` files that we have for the old tests (now `graph local`).